core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2865,8 +2865,13 @@ private void getExpressionConditionsAsList(Condition condition, List<ExpressionC
                 do {
                     List<String> tempUserList = new ArrayList<>();
                     answer = ldapContext.search(escapeDNForSearch(searchBase), searchFilter, searchControls);
+                    /*
+                    The size of the expression condition is passed, since the up coming steps needs to verify
+                    whether this is a multi attribute scenario or single attribute scenario.
+                     */
                     if (answer.hasMore()) {
-                        tempUserList = getUserListFromSearch(isGroupFiltering, returnedAttributes, answer);
+                        tempUserList = getUserListFromSearch(isGroupFiltering, returnedAttributes, answer,
+                                expressionConditions.size());
```

Comment:
If the scenario is identifiable from the expression list size why not identify that at this point and pass a boolean for the scenario or an identifier for the scenario from this point rather than passing a size

Reply:
Fixed: b4a1c359b2cdcd987db1766b9580db2a011d7f28